### PR TITLE
[incubator/cassandra] Add podAntiAffinity

### PR DESCRIPTION
**Pull request by TheMeier**
Add podAntiAffinity to incubator/cassandra with a default to a "soft" requirement  -> `preferredDuringSchedulingIgnoredDuringExecution `

### CODE REVIEW

File: incubator/cassandra/templates/cassandra-statefulset.yaml [hunk 1]
```diff
@@ -60,6 +60,29 @@ spec:
 {{- if .Values.selector }}
 {{ toYaml .Values.selector | indent 6 }}
 {{- end }}
+      serviceAccountName: {{ template "cassandra.fullname" . }}
```
Comment (unguiculus) [inline incubator/cassandra/templates/cassandra-statefulset.yaml]: This adds a service account that may not exist.

Comment (TheMeier) [reply incubator/cassandra/templates/cassandra-statefulset.yaml]: right actually i did remove that in my local setup already, i will push an update in a few minutes

**Comment by unguiculus** [comment]
For full flexibility, it'd be better to do it like this:

https://github.com/kubernetes/charts/blob/27b97dffdb393d6c9e7f65acb673547daa3bf38e/stable/minio/templates/minio_deployment.yaml#L31-L34

**Pull request merged**